nbgrader/apps/autogradeapp.py

```diff
@@ -171,7 +193,12 @@ def _init_preprocessors(self):
             preprocessors = self.autograde_preprocessors
 
         for pp in preprocessors:
-            self.exporter.register_preprocessor(pp)
+            if self.execute:
+                self.exporter.register_preprocessor(pp)
+            else:
+                # If execute is False, don't run notebooks or clear output
+                if pp not in (Execute, ClearOutput):
+                    self.exporter.register_preprocessor(pp)
```

Comment:
Seem like `no-execute` could just be an alias for `Execute.enabled=False` and `ClearOutput .enabled=False` instead of a convoluted logic. No ?